website/public/js/controllers/challengesCtrl.js

```diff
@@ -88,6 +88,51 @@ habitrpg.controller("ChallengesCtrl", ['$rootScope','$scope', 'Shared', 'User',
     };
 
     /**
+     * Clone
+     */
+    $scope.clone = function(challenge) {
+      //We need to clone habits, dailys, todos, and rewards. They each need a new id and entry in the database
+      var clonedTasks = {
+        habit: [],
+        daily: [],
+        todo: [],
+        reward: []
+      };
+
+      function cloneTask(taskToClone, index, array) {
+        var tmpTask = {};
+        for( var property in taskToClone ) {
+          if ( property !== "_id" && property !== "id" ) {
+            tmpTask[property] = taskToClone[property];
```

Comment:
Aw, man. I think you are right. I definitely overthought this. I would need to delete both id properties, I believe. I'll test it in a few minutes.